src/components/activities/pharmacyActivity/pharmaceutical/components/PharmacyCellContent/CellContent.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { FC } from "react";
+import { Typography } from "@mui/material";
+import { Warning, EventBusy } from "@mui/icons-material";
+
+interface ICellContentProps {
+  value: any;
+  keyName: string;
+  row: any;
+  hasExpiringLotThisMonth: () => boolean;
+}
+
+const CellContent: FC<ICellContentProps> = ({
+  value,
+  keyName,
+  row,
+  hasExpiringLotThisMonth,
+}) => {
+  return (
+    <div className="cell-content">
+      {row.stock === 0 && keyName === "pharmaceutical" && (
+        <Typography component="span" className="stock-zero-badge">
+          0
+        </Typography>
+      )}
+
+      {row.stock < row.criticalValue &&
+        hasExpiringLotThisMonth() &&
+        keyName === "pharmaceutical" && (
+          <>
+            <Warning className="icon-warning" />
+            <EventBusy className="icon-expiry" />
+          </>
+        )}
+
+      {hasExpiringLotThisMonth() &&
```

Comment:
```suggestion
      {hasExpiringLotThisMonth &&
```

Once you'll move the logic in this component and use useMemo as mentionned above, this expression will no longer be available.